commands/scanemotes.js

```diff
@@ -2,158 +2,170 @@
-
-                                if (!(emoteID in statsWithoutBots)) statsWithoutBots[emoteID] = 0;
-                                statsWithoutBots[emoteID] += msg.reactions.get(emoteTag).count;
-
-                                // I don't know why this collection always appears as empty.
-                                msg.reactions.get(emoteTag).users.forEach(user => {
-                                    if (user.bot) statsWithoutBots[emoteID]--;
-                                });
+                        for(let reaction of msg.reactions.values()) {
+                            const emoteID = reaction.emoji.id;
+                            let continueReactionLoop = true;
+                            let lastUserID;
+                            let userReactions = 0;
+                            let botReactions = 0;
+
+                            if (emoteID in stats) {
+                                // Unfortunately, just like fetchMessages, fetchUsers is limited to requesting 100 maximum.
+                                // There is a simple count property on a reaction, but that doesn't separate users from bots.
+                                // So instead, I'll use that property to check for inconsistencies.
+                                while (continueReactionLoop) {
+                                    // After logging fetchUsers, it seems like the order is strictly numerical. As long as that stays consistent, this should work fine.
+                                    const users = await reaction.fetchUsers(100, {
+                                        after: lastUserID
+                                    });
+
+                                    if (users.size > 0) {
+                                        for(let user of users.values()) {
+                                            if (user.bot) {
+                                                stats[emoteID].bots++;
+                                                botReactions++;
+                                            } else {
+                                                stats[emoteID].users++;
+                                                totalUserEmoteUsage++;
+                                                userReactions++;
+                                            }
+
+                                            lastUserID = user.id;
+                                        };
+                                    } else {
+                                        // Then halt the loop and send warnings of any inconsistencies.
+                                        continueReactionLoop = false;
+
+                                        if (reaction.count !== userReactions + botReactions) {
+                                            console.warn(`[Channel: ${channel.id}, Message: ${msg.id}] A reaction count of ${reaction.count} was expected but was given ${userReactions} user reactions and ${botReactions} bot reactions.`);
```

Comment:
Perhaps using the logger module would be better to keep the unity between everything (`client.logger.log`)

That's for you to decide, though

Reply:
```suggestion
                                            client.logger.warn(`[Channel: ${channel.id}, Message: ${msg.id}] A reaction count of ${reaction.count} was expected but was given ${userReactions} user reactions and ${botReactions} bot reactions.`);
```